### Домашнее задание №3

***Описательная статистика. Качественные и количественные характеристики популяции. Графическое представление данных***

***1) Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.***

In [1]:
import pandas as pd
from math import sqrt

salary=pd.DataFrame([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])
#salary = [100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150]
mu = salary[0].sum()/salary[0].count()
print(f'Математическое ожидание = {mu}')
s = 0
for value in salary[0]:
    s+= (value-mu)**2
print(f"Среднее квадратичное отклонение: {sqrt(s/salary[0].count())}")
print(f"Смещенная оценка дисперсии: {s/salary[0].count()}")
print(f"Несмещенная оценка дисперсии: {s/(salary[0].count()-1)}")


Математическое ожидание = 65.3
Среднее квадратичное отклонение: 30.823854398825596
Смещенная оценка дисперсии: 950.11
Несмещенная оценка дисперсии: 1000.1157894736842


***2) В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?***

Опишем все возможные варианты событий:
<ol>
<li>два мяча из первого ящика белые, один мяч из второго ящика белый и три чёрных;
<li>два мяча из первого ящика чёрные, три мяча из второго ящика белые и один чёрный;
<li>один мяч из первого ящика белый и второй мяч из первого ящика чёрный, два мяча из второго ящика белые и два мяча из второго ящика чёрные.
</ol>


Вероятность что из первого ящика вытащат два белых мяча:

In [2]:
p1 = (5/8)*(4/7)

Вероятность того, что один мяч из второго ящика белый а три мяча из второго ящика чёрные, для нас порядок не важен, будет суммой вероятностей четырех исходов
<li>Cначала один белый потом три чёрных;
<li>Сначала один черный, потом один белый, затем два черных;
<li>Сначала два черных, потом один белый, затем один черный;
<li>Сначала три черных, потом один белый.

In [3]:
p2 = (5/12)*(7/11)*(6/10)*(5/9) + (7/12)*(5/11)*(6/10)*(5/9) + (7/12)*(6/11)*(5/10)*(5/9) + (7/12)*(6/11)*(5/10)*(5/9)

In [4]:
P1 = p1*p2
P1

0.12626262626262624

Вероятность того, что из первого ящика будет вытащено 2 черных мяча:

In [5]:
p1 = (3/8)*(2/7)

Вероятность того, что три мяча из второго ящика белые, а один мяч из второго ящика чёрный, для нас порядок не важен, будет суммой вероятностей четырех исходов:
<li>Cначала три белых, потом один чёрный;
<li>Сначала два белых, потом один черный, затем один белый;
<li>Сначала один белый, потом один черный, затем два белых;
<li>Сначала один черный, потом три белых. 

In [6]:
p2 = (5/12)*(4/11)*(3/10)*(7/9) + (5/12)*(4/11)*(7/10)*(3/9) + (5/12)*(7/11)*(4/10)*(3/9) + (7/12)*(5/11)*(4/10)*(3/9)

In [7]:
P2 = p1*p2
P2

0.01515151515151515

Вероятность того, что один мяч из первого ящика белый, а второй мяч из первого ящика чёрный, для нас порядок не важен, будет суммой вероятностей двух исходов:
<li>первый мяч из первого ящика белый, а второй мяч черный
<li>первый мяч из первого ящика черный, а второй мяч белый

In [8]:
p1 = (5/8)*(3/7)+(3/8)*(5/7)

Вероятность того, что два мяча из второго ящика белые и два мяча из второго ящика чёрные, для нас порядок не важен, будет суммой вероятностей шести исходов:
<li>первые два меча белых, затем два черных;
<li>один мяч белый, два черных, один белый;
<li>два черных, потом два белых;
<li>первый мяч черный, второй белый, третий черный, четвертый белый; 
<li>первый мяч черный, второй и третий белый, четвертый черный; 
<li>первый мяч белый, второй черный, третий белый, четвертый черный.


In [9]:
p2 = (5/12)*(4/11)*(7/10)*(6/9)+(5/12)*(7/11)*(6/10)*(4/9)+(7/12)*(6/11)*(5/10)*(4/9)+(7/12)*(5/11)*(6/10)*(4/9)+(7/12)*(5/11)*(4/10)*(6/9)+(5/12)*(7/11)*(4/10)*(6/9)

In [10]:
P3 = p1*p2
P3

0.22727272727272727

Итоговая вероятнось что три мяча будут белых равна сумме вероятностей

In [11]:
P = P1 + P2 + P3

In [12]:
P

0.3686868686868686

#### Ответ: P = 0.3686868686868686 ####

***3) На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.***

Если был произведен выстрел, вероятность того что выстрелил любой из спортсменов равна $\frac{1}{3}$

Искомым событием $A$ будем считать попадание в мишень для каждого из стрелка, а событиями $B_1, B_2$ и $B_3$ — что выстрел совершил первый, второй или третий спортсмен:

Вероятности попаданий в мешень для каждого из стрелков:<br>
$P(A\:|\:B_1) = 0.9$

$P(A\:|\:B_2) = 0.8$

$P(A\:|\:B_3) = 0.6$

Вероятность того, что стрелял первый спортсмен и попал в мишень, расчитывается след.образом:

$P(B_1\:|\:A) = \frac{P(B_1) \cdot P(A\:|\:B_1)}{P(A)}$

$P(B_1\:|\:A) = \frac{\frac{1}{3} \cdot 0.9}{\frac{1}{3} \cdot 0.9 + \frac{1}{3} \cdot 0.8 + \frac{1}{3} \cdot 0.6} = \frac{9}{23}$

Вероятность того, что стрелял второй спортсмен и попал в мишень, расчитывается след.образом:

$P(B_2\:|\:A) = \frac{P(B_1) \cdot P(A\:|\:B_1)}{P(A)}$

$P(B_2\:|\:A) = \frac{\frac{1}{3} \cdot 0.8}{\frac{1}{3} \cdot 0.9 + \frac{1}{3} \cdot 0.8 + \frac{1}{3} \cdot 0.6} = \frac{8}{23}$

Вероятность того, что стрелял третий спортсмен и попал в мишень находится по формуле:

$P(B_3\:|\:A) = \frac{P(B_3) \cdot P(A\:|\:B_3)}{P(A)}$

$P(B_3\:|\:A) = \frac{\frac{1}{3} \cdot 0.6}{\frac{1}{3} \cdot 0.9 + \frac{1}{3} \cdot 0.8 + \frac{1}{3} \cdot 0.6} = \frac{6}{23}$

#### Ответ: Искомые вероятности 1) $\frac{8}{33}$; 2) $\frac{7}{33}$; 3) $\frac{18}{33}$ ####

***5) Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?***

а) Вероятность того, что в первый месяц выйдут из строя все детали:

In [13]:
0.1*0.2*0.25

0.005000000000000001

б) Вероятность того, что в первый месяц выйдут из строя только 2 детали равна сумме вероятностей событий:
<li>первая и вторая детали вышли из строя, а третья нет;
<li>первая и третья детали вышли из строя, а вторая нет;
<li>вторая и третья детали вышли из строя, а первая нет.

In [14]:
0.1*0.2*(1-0.25)+0.1*(1-0.2)*0.25+(1-0.1)*0.2*0.25

0.08000000000000002

в) Вероятность того, что в первый месяц выйдет из строя хоть одна деталь (противоположна вероятности что не выйдет ни одна деталь из строя):

In [15]:
1 - (1-0.1)*(1-0.2)*(1-0.25)

0.45999999999999996

г) Вероятность что в первый месяц выйдут из строя от одной до двух деталей, является противоположным сумме вероятностей событий:
<li> что в первый месяц не выйдет из строя ни одна деталь;
<li> что в первый месяц выйдут из строя все 3 детали.<br>
Вероятность равна:

In [16]:
1 - ((1-0.1)*(1-0.2)*(1-0.25) + 0.1*0.2*0.25)

0.45499999999999996

#### Ответ: а) 0.005; б) 0.08; в) 0.46; г) 0.455 ####